### PR TITLE
add VacuumPermittivity and fix adding user values to material editor …

### PR DESCRIPTION
**Pull request by dietmarczekay**
…in FEM

Thank you for creating a pull request to contribute to FreeCAD! To ease integration, please confirm the following:

- [x ] Branch rebased on latest master `git pull --rebase upstream master`

- adding vacuum permittivity as unit and some fixes for adding user properties to material editor

### CODE REVIEW

**Comment by wwmayer** [comment]
@berndhahnebach is it OK for you?

**Comment by justnope** [comment]
I was having the same problem with adding user defined values. I'm new to the code base but could someone if the following lines also need to be changed?

https://github.com/FreeCAD/FreeCAD/blob/679f7e91c75c42db3ea22646a85ff7f7d491aad7/src/Mod/Material/importFCMat.py#L165

https://github.com/FreeCAD/FreeCAD/blob/679f7e91c75c42db3ea22646a85ff7f7d491aad7/src/Mod/Material/materialtools/cardutils.py#L255-L256

https://github.com/FreeCAD/FreeCAD/blob/679f7e91c75c42db3ea22646a85ff7f7d491aad7/src/Mod/Material/materialtools/cardutils.py#L291-L292

File: src/Mod/Material/Templatematerial.yml [hunk 1]
```diff
@@ -265,7 +270,7 @@
       Type: 'String'
       URL: ''
       Description: "Vendor of the material"
-- UserDefined:
+- User defined:
```
Comment (yorikvanhavre) [inline src/Mod/Material/Templatematerial.yml]: Is this really safe? In any case it most certainly would need further changes. I suggest you revert this line, and make another PR with it if applicable

**Comment by dietmarczekay** [comment]
No problem, I just used the same way as for "-Vector rendering" some lines above

**Comment by berndhahnebach** [comment]
I missed this PR. I will have a look. 

@dietmarczekay: would have been good to start the PR heading with Material: or FEM:  This is what my eyes are trained for, but nevermind ...

File: src/Mod/Material/Templatematerial.yml [hunk 2]
```diff
@@ -155,6 +155,11 @@
       Type: 'Quantity'
       URL: 'https://en.wikipedia.org/wiki/Volumetric_thermal_expansion_coefficient'
       Description: "Thermal expansion coefficient (linear) in [FreeCAD ThermalExpansionCoefficient unit]"
+- Electrical:
+    RelativePermittivity:
+      Type: 'Float'
+      URL: 'https://en.wikipedia.org/wiki/Relative_permittivity'
+      Description: "a material property, expressing the ratio to the permittivity of the Vaccum"
```
Comment (berndhahnebach) [inline src/Mod/Material/Templatematerial.yml]: typo: Vacuum

**Comment by berndhahnebach** [comment]
What was the origin reason to change `UserDefined `to `User Defined` ?  We may fix it at another place?

**Comment by yorikvanhavre** [comment]
Thanks for the change @dietmarczekay It's cleaner that way so this PR only applies to VacuumPermittivity.

In general, regarding these CamelCase values, my own position would be to do the same way as in the FreeCAD properties panel: Internally, properties are stored as "CamelCase" everywhere. That's more guarantee for possible future parsing operations that might not swallow spaces. We never know what the future could do with our material cards... However, everywhere that value is presented to the user in interfaces, we can split and show as "Camel Case".

**Comment by dietmarczekay** [comment]
Thanks for your explanation.
The change was needed because in Templatematerial.yml was a key value `Userdefined`. So MaterialEdiotr.py didn't find that in line 297 and we were not able to add custom values. First I changed TemplateMaterial.py to `User Defined` and after Yorik's suggestion to `UserDefined`

Dietmar.

**Comment by berndhahnebach** [comment]
What is the difference between RelativePermittivity and VacuumPermittivity?

**Comment by dietmarczekay** [comment]
Relative permittivity is a material constant, relative to the vacuum. Describes the influence from material to the electric field...
